### PR TITLE
Recommend Channel<T> as async alternative to BlockingCollection<T>

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Addresses https://github.com/dotnet/runtime/issues/47587.

### CODE REVIEW

**Comment by eiriktsarpalis** [comment]
cc @layomia @gewarren

File: xml/System.Collections.Concurrent/BlockingCollection`1.xml [hunk 1]
```diff
@@ -100,6 +100,8 @@
...
   
  Do not modify the underlying collection directly. Use <xref:System.Collections.Concurrent.BlockingCollection%601> methods to add or remove elements. The <xref:System.Collections.Concurrent.BlockingCollection%601> object can become corrupted if you change the underlying collection directly.  
+
+ <xref:System.Collections.Concurrent.BlockingCollection%601> has not been designed with asynchronous access in mind; consider using <xref:System.Threading.Channels.Channel%601> instead if your application requires async producer/consumer scenaria.
```
Comment (gewarren) [inline xml/System.Collections.Concurrent/BlockingCollection`1.xml:104]:
```suggestion
<xref:System.Collections.Concurrent.BlockingCollection%601> was not designed with asynchronous access in mind. If your application requires asynchronous producer/consumer scenarios, consider using <xref:System.Threading.Channels.Channel%601> instead.
```

**Comment by gewarren** [comment]
Should we wait for Layomi or go ahead and merge?

**Comment by eiriktsarpalis** [comment]
I think we can merge.

**Pull request merged**